prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -1287,9 +1288,12 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
     fn encode_loan_places(&self, loan_places: &LoanPlaces<'tcx>) -> (vir::Expr, vir::Expr, bool) {
         debug!("encode_loan_places '{:?}'", loan_places);
         // will panic if attempting to encode unsupported type
-        let (expiring_base, expiring_ty, _) = self.mir_encoder.encode_place(&loan_places.dest).unwrap();
+        let (expiring_base, pre_stmts, expiring_ty, _) = self.encode_place(&loan_places.dest).unwrap();
+        assert!(pre_stmts.is_empty(), "Unexpected encoding pre-statements: {:?}", pre_stmts);
+
         let encode = |rhs_place| {
-            let (restored, _, _) = self.mir_encoder.encode_place(rhs_place).unwrap();
+            // TODO: pre_stmts here as well..
+            let (restored, pre_stmts, _, _) = self.encode_place(rhs_place).unwrap();
```

Comment:
Adding `assert!(pre_stmts.is_empty(), "TODO");` as you did in other places sounds better.